hphp/runtime/vm/jit/vasm-simplify.cpp

```diff
@@ -375,22 +375,35 @@ bool simplify(Env& env, const copyargs& inst, Vlabel b, size_t i) {
 }
 
 bool simplify(Env& env, const movzlq& inst, Vlabel b, size_t i) {
-  if (arch() != Arch::X64) return false;
-  auto const def_op = env.def_insts[inst.s];
-
-  // Check if `inst.s' was defined by an instruction with Vreg32 operands, or
-  // movzbl{} in particular (which lowers to a movl{}).
-  if (width(def_op) != Width::Long &&
-      def_op != Vinstr::movzbl) {
-    return false;
+  switch (arch()) {
```

Comment:
I considered the explicit list using a switch to be safer in case that another architectures comes along some day.  Note that the only thing that comes to mind is MIPS64, so this is probably a nit-pick by me.

The benefits of listing this explicitly in a switch-statement are:

- any new architectures won't pick this up automatically
- the compiler will generate a warning if not all architectures are listed

That said: if you want it changed, I'll change it...